src/dune_rules/melange_rules.ml

```diff
@@ -78,7 +78,7 @@ let build_js ~loc ~dir ~pkg_name ~module_system ~dst_dir ~obj_dir ~sctx
        ])
 
 let add_rules_for_entries ~sctx ~dir ~expander ~dir_contents ~scope
-    ~compile_info ~target_dir (mel : Melange_stanzas.Emit.t) =
+    ~compile_info ~target_dir ~mode (mel : Melange_stanzas.Emit.t) =
```

Comment:
If `Rule.Mode.t` was derived just before calling `Super_context.add_rule`, we'd avoid the param here.

Reply:
I'd rather compute the mode once than avoid passing an argument.